src/hssm/hssm.py

```diff
@@ -678,7 +804,25 @@ def sample_prior_predictive(
             ``InferenceData`` object with the groups ``prior``, ``prior_predictive`` and
             ``observed_data``.
         """
-        return self.model.prior_predictive(draws, var_names, omit_offsets, random_seed)
+        prior_predictive = self.model.prior_predictive(
+            draws, var_names, omit_offsets, random_seed
+        )
+
+        prior_predictive.add_groups(posterior=prior_predictive.prior)
+        self.model.predict(prior_predictive, kind="mean", inplace=True)
+
+        # clean
+        prior_predictive.prior = prior_predictive.posterior
+        del prior_predictive.posterior
+
+        if self._inference_obj is None:
+            self._inference_obj = prior_predictive
+        else:
+            self._inference_obj.extend(prior_predictive)
+
+        # clean up `rt,response_mean` to `v`
+        self._drop_parent_str_from_idata(self._inference_obj)
+        return self._inference_obj
```

Comment:
I don't quite understand the logic here. `sample_prior_predictive` can be called before any sampling is done, right? Then why should we attach it to self._inference_obj? Then when you call `sample`, this object will be overridden after sampling. It doesn't quite work with some of the logic in the rest of the code either.

Reply:
had discussion offline on this one.